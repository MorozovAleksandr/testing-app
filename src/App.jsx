import { Link, Route, Routes } from "react-router-dom";

import MainPage from "./pages/MainPage/MainPage";
import AboutPage from "./pages/AboutPage/AboutPage";
import ErrorPage from "./pages/ErrorPage/ErrorPage";

import "./App.css";
import Users from "./pages/UsersPage/Users";
import UsersDeatailsPage from "./pages/UsersDeatailsPage/UsersDeatailsPage";

import "./App.css";

const App = () => {
  return (
    <>
      <div className={"links"}>
        <Link className={"link"} to="/" data-testid="main-link">
          main
        </Link>
        <Link className={"link"} to="about" data-testid="about-link">
          about
        </Link>
        <Link className={"link"} to="users" data-testid="users-link">
          users
        </Link>
      </div>
      <Routes>
        <Route path="/" element={<MainPage />} />
        <Route path="/about" element={<AboutPage />} />
        <Route path="/users" element={<Users />} />
        <Route path="/users/:id" element={<UsersDeatailsPage />} />
        <Route path="/*" element={<ErrorPage />} />
      </Routes>
    </>
  );
};

export default App;
