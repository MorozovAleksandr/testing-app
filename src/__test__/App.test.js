import { render, screen, fireEvent } from "@testing-library/react";
import App from "../App";

describe("App", () => {
  test("Elements", () => {
    render(<App />);
    const helloWorldEl = screen.getByText(/Hello world/i);
    const btnEl = screen.getByRole("button");
    const inputEl = screen.getByPlaceholderText(/some text.../i);
    expect(helloWorldEl).toBeInTheDocument();
    expect(btnEl).toBeInTheDocument();
    expect(inputEl).toBeInTheDocument();
    expect(inputEl).toMatchSnapshot();
    // screen.debug();
  });
  
  test("Click event", () => {
    render(<App />);
    const toggleBtn = screen.getByTestId('toggle-btn');
    expect(screen.queryByTestId('toggle-div')).toBeNull();
    fireEvent.click(toggleBtn)
    expect(screen.queryByTestId('toggle-div')).toBeInTheDocument();
    fireEvent.click(toggleBtn)
    expect(screen.queryByTestId('toggle-div')).toBeNull();
    // screen.debug();
  });
  
  test("Input event", () => {
    render(<App />);
    const inputEl = screen.getByPlaceholderText(/some text.../i);
    expect(screen.queryByTestId('input-value')).toContainHTML('');
    fireEvent.input(inputEl, {
      target: {
        value: '1337',
      }
    });
    expect(screen.queryByTestId('input-value')).toContainHTML('1337');
    // screen.debug();
  });
});
