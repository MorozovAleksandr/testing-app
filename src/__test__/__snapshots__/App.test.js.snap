// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Elements 1`] = `
<input
  placeholder="some text..."
  type="text"
  value=""
/>
`;
